crates/indexer/src/models/token_models/collection_datas.rs

```diff
@@ -73,13 +81,16 @@ impl CollectionData {
         };
         if let Some(collection_data) = maybe_collection_data {
             let table_handle = table_item.handle.to_string();
-            let creator_address = table_handle_to_owner
-                            .get(&TableMetadataForToken::standardize_handle(&table_handle))
-                            .map(|table_metadata| table_metadata.owner_address.clone())
-                            .context(format!(
-                                "version {} failed! collection creator resource was missing, table handle {} not in map {:?}",
-                                txn_version, TableMetadataForToken::standardize_handle(&table_handle), table_handle_to_owner,
-                            ))?;
+            let maybe_creator_address = table_handle_to_owner
+                .get(&TableMetadataForToken::standardize_handle(&table_handle))
+                .map(|table_metadata| table_metadata.owner_address.clone());
+            let creator_address = match maybe_creator_address {
+                Some(ca) => ca,
+                None => Self::get_collection_creator(conn, &table_handle).context(format!(
+                    "Failed to get collection creator for table handle {}, txn version {}",
+                    table_handle, txn_version
+                ))?,
+            };
```

Comment:
Please put a comment above this describing what it's doing

Reply:
Cool.